external/radiation/tests/test_preprocessing.py

```diff
@@ -0,0 +1,39 @@
+from radiation.preprocessing import _flatten, _merge
+import pytest
+import numpy as np
+
+ARR = np.array(3)
+
+
+@pytest.mark.parametrize(
+    ["in_dict", "flattened_dict", "raises_type_error"],
+    [
+        pytest.param({"a": ARR}, {"a": ARR}, False, id="flat"),
+        pytest.param({"a": {"b": ARR}}, {"a_b": ARR}, False, id="nested"),
+        pytest.param({"a": 3}, None, True, id="type_error"),
+    ],
+)
+def test__flatten(in_dict, flattened_dict, raises_type_error):
```

Comment:
Nit. It would be a bit less complex and readable to split the error raising test into a new test function.